records_mover/db/postgres/postgres_copy_options.py

```diff
@@ -0,0 +1,362 @@
+from ...utils import quiet_remove
+from ...records.hints import cant_handle_hint
+from typing import Dict, Set
+from ...records.load_plan import RecordsLoadPlan
+from records_mover.records.types import RecordsHints
+from ...records.records_format import DelimitedRecordsFormat
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+PostgresCopyOptions = Dict[str, object]
+
+# https://www.postgresql.org/docs/9.2/sql-copy.html
+
+
+def needs_csv_format(hints: RecordsHints) -> bool:
+    # This format option is used for importing and exporting the Comma
+    # Separated Value (CSV) file format used by many other programs,
+    # such as spreadsheets. Instead of the escaping rules used by
+    # PostgreSQL's standard text format, it produces and recognizes
+    # the common CSV escaping mechanism.
+
+    # The values in each record are separated by the DELIMITER
+    # character. If the value contains the delimiter character, the
+    # QUOTE character, the NULL string, a carriage return, or line
+    # feed character, then the whole value is prefixed and suffixed by
+    # the QUOTE character, and any occurrence within the value of a
+    # QUOTE character or the ESCAPE character is preceded by the
+    # escape character. You can also use FORCE_QUOTE to force quotes
+    # when outputting non-NULL values in specific columns.
+    if hints['header-row'] or (hints['quoting'] is not None):
+        return True
+
+    return False
+
+
+def postgres_copy_options_text(unhandled_hints: Set[str],
+                               hints: RecordsHints,
+                               fail_if_cant_handle_hint: bool,
+                               postgres_options: PostgresCopyOptions) -> None:
+    # FORMAT
+    #
+    #  Selects the data format to be read or written: text, csv (Comma
+    #  Separated Values), or binary. The default is text.
+    #
+    postgres_options['format'] = 'text'
+
+    # Backslash characters (\) can be used in the COPY data to quote
+    # data characters that might otherwise be taken as row or column
+    # delimiters. In particular, the following characters must be
+    # preceded by a backslash if they appear as part of a column
+    # value: backslash itself, newline, carriage return, and the
+    # current delimiter character.
+    if hints['escape'] is None:
+        cant_handle_hint(fail_if_cant_handle_hint, 'escape', hints)
+    else:
+        quiet_remove(unhandled_hints, 'escape')
+
+    # OIDS
+    #
+    #  Specifies copying the OID for each row. (An error is raised if
+    #  OIDS is specified for a table that does not have OIDs, or in
+    #  the case of copying a query.)
+    #
+
+    # TODO: OIDS
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker